javarosa/src/main/java/org/javarosa/xpath/expr/XPathFuncExpr.java

```diff
@@ -660,6 +659,20 @@ public static Double toDouble(Object o) {
         }
     }
 
+    private static Double toNumeric_inclusiveOfDatestrings(Object o) {
+        Double d = toNumeric(o);
+        if (Double.isNaN(d.doubleValue())) {
+            o = unpack(o);
+            if (o instanceof String) {
+                Date dateFromString = DateUtils.parseDate((String)o);
+                if (dateFromString != null) {
+                    return new Double(DateUtils.daysSinceEpoch(dateFromString));

```

Comment:
For cohesiveness, can we make sure that the process by which a string turns into a date, and a date turns into a double are following the same code paths as they would be normally? IE: seems like we should be doing 

if(NaN) { 
try
toNumeric(toDate(val))
catch()
}

So that if there's any complexity (your unpack() call is a good example), that complexity is compartmentalized.